src/ParametricOptInterface.jl

```diff
@@ -1570,6 +1566,11 @@ function MOI.optimize!(model::Optimizer)
         set_quadratic_product_in_obj!(model)
     end
     MOI.optimize!(model.optimizer)
+    if MOI.get(model, MOI.DualStatus()) == MOI.NO_SOLUTION &&
+       model.evaluate_duals
+        @warn "Dual solution not available, setting evaluate_duals to false"
+        model.evaluate_duals = false
+    end
     if model.evaluate_duals
         calculate_dual_of_parameters(model)
     end
```

Comment:
```suggestion
    if MOI.get(model, MOI.DualStatus()) == MOI.NO_SOLUTION &&
       model.evaluate_duals
        @warn "Dual solution not available, ignoring `evaluate_duals`"
    elseif model.evaluate_duals
        calculate_dual_of_parameters(model)
    end
```

Reply:
main problem with the implicit change is that it will be changed independed of what the user set, so was confusing. This way is better, because if the problem is modified and there is dual solution then it will be computed.